### PR TITLE
ROX-12977: Update to align with current Red Hat Insights starter app

### PR DESCRIPTION
**Pull request by bradr5**
## Description
Update everything to align with current Red Hat Insights starter app

This satisfies the following tasks:

- ROX-12958
- ROX-14786
- ROX-12977

## Checklist (Definition of Done)

- [ ] Unit and integration tests added
- [ ] Added test description under `Test manual`
- [ ] Documentation added if necessary
- [ ] CI and all relevant tests are passing
- [ ] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [ ] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.

## Test manual

**TODO:**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+language: node_js
+sudo: required
+notifications:
+  email: false
+  slack:
+    rooms:
+      # insights-bot channel on [Ansible Slack](https://ansible.slack.com)
+      - secure: juuizlfiQolXbqy4L2nfODB1K8xiSGvUrzsT3XeXIsQqX7yCYVvnRUssKLbqgLM9m85kgXJk1ef0qwjJoliNqGcrALTZLHPTkxWwTnV7tPEVS/4uQPSyy+DDHDwrTzCrF3qhCqyKJ6vB18z22qm3qLgStmNj4p1ZLznsdJR/u7U3spJ+WLL70QSNHCQy/5QjRDDiR3oqWdXjhVcUCeV+v/2CtfqK3dqYtzCL/b7z8lm/YbV6w1Rd8mwzTgLstbzjrvR/gq2IPALzzJTq1S8lXtV7ZtbYz4bjB8RNCRZFJrjBUH4CqD1oUskit4RRe+ilVs8qNQh0Tt3whsOdbJoRXJn2B/NVxppp2WGlODdsuMgfNh5iBocHKN3Qo4mONbstnHFG99Hr7L8GbUugJHEQlrZn1xRyO9ulyVp+hFp8/F088NNyhnd3+lERUPA5/4+HPQh6Ea+PyKq2TR41VoDGcRDBS4VY/6PfLEJzxEBjwRAAemfvtOFTUu2EocuAllPX1vMA4b1j7Wm+Bs7wft6nEcVdftVanfAgx4dNksnaS8zN9II8rdtBci2yXYJ6J+e+REDn/2IrHaUwr0N9BojYc0ePUElDcygW4Kk1oGklHp681j2zhi1Gt7rbuae1C9fD4QHnryx59uVH5NXqDsXI5EbwTJE3zvzRz9SzmyDqV5A=
```
Comment (ryelo) [inline .travis.yml:8]: This can be removed. It's getting deprecated next week.

**Review by sachaudh — COMMENTED** [review]
Oh nice, we get Typescript with these changes. Let me do a more thorough check since there are a lot of changes going into this PR

File: src/routes/NoPermissionsPage/NoPermissionsPage.tsx [hunk 2]
```diff
@@ -1,10 +1,9 @@
 import React, { useEffect } from 'react';
-import { withRouter } from 'react-router-dom';
 
 import { Main } from '@redhat-cloud-services/frontend-components/Main';
 import { NotAuthorized } from '@redhat-cloud-services/frontend-components/NotAuthorized';
 
-function NoPermissionsPage() {
+const NoPermissionsPage = () => {
```
Comment (vjwilson) [inline src/routes/NoPermissionsPage/NoPermissionsPage.tsx:6]: nit: Some of components are still defined with `function` keyword, but this and `OopsPage.tsx` use arrow function definition.

While rare, there are some weird edge cases that can occur with top-level arrow functions, so I'd prefer we leave them with the `function` keyword.

File: src/routes/OopsPage/OopsPage.tsx [hunk 3]
```diff
@@ -1,10 +1,9 @@
 import React, { useEffect } from 'react';
-import { withRouter } from 'react-router-dom';
 
 import { Main } from '@redhat-cloud-services/frontend-components/Main';
 import { Unavailable } from '@redhat-cloud-services/frontend-components/Unavailable';
 
-function OopsPage() {
+const OopsPage = () => {
```
Comment (vjwilson) [inline src/routes/OopsPage/OopsPage.tsx:6]: nit: Prefer `function` keyword for top-level functions

**Review by vjwilson — CHANGES_REQUESTED** [review]
I'm fine with this overall, but we need to update the README, too, to reflect the new dev process.

(plus, a couple of nits)

**Review by vjwilson — APPROVED** [review]

**Pull request merged**